python/orca/src/bigdl/orca/learn/utils.py

```diff
@@ -208,11 +210,25 @@ def convert_elem(elem):
                 return elem.tolist()
 
     def combine(pair):
-        return Row(*([pair[0][col] for col in pair[0].__fields__] +
-                     [convert_elem(pair[1])]))
+        if not output_cols:
+            if isinstance(pair[1], (list, tuple)) and len(pair[1]) == 1:
+                return Row(*([pair[0][col] for col in pair[0].__fields__] +
+                             convert_elem(pair[1])))
```

Comment:
row should be <[input_1], [input_2], [DenseVector]>.

Reply:
shouldn't it be Row(x1, x2, output)? * will do the unpack for lists.

Reply:
that depends o, if pair[1] is ndarray, output (Dense Vector) is fine, if pair[1] is a list of ndarray, we should convert it to [Dense Vector].

Reply:
so why should we discuss the case for a singleton list?